docs/source/models.rst

```diff
@@ -307,6 +332,16 @@ be made compatible, MLflow will raise an error.
 For models with tensor-based signatures, type checking is strict (i.e an exception will be thrown if
 the input type does not match the type specified by the schema).
 
+Params Type and Shape Enforcement
+"""""""""""""""""""""""""""""""""
+The params types and shapes are checked against the signature.
+
+MLflow verifies the compatibility of each parameter provided during inference by comparing its type and shape 
+with those specified in the signature. Scalar values should have a shape of ``None``, while list values should have 
+a shape of ``(-1,)``. If the parameter's type or shape is incompatible, an exception will be raised. 
+Additionally, the value of the parameter is validated against the specified type in the signature. We attempt 
+to convert the value to the specified type, and if this conversion fails, an MlflowException will be raised.
+
```

Comment:
Can we provide a list with generic examples of what parameter types are supported?
Can there also be an explanation added of why this feature exists and how it can be used as an intro to a new section above this? This is a pretty important and long asked-for feature that I think another `"""""` section is warranted to explain that inference parameters are supported now :)